core-web/apps/dotcms-ui/src/app/shared/directives/dot-show-hide-feature/dot-show-hide-feature.directive.ts

```diff
@@ -0,0 +1,76 @@
+import { Component, Directive, Input, OnInit, TemplateRef, ViewContainerRef } from '@angular/core';
+
+import { DotPropertiesService } from '@dotcms/data-access';
+
+/**
+ * Structural directive to give the ability of show or hide a component
+ * based on a feature flag. This directive can receive an alternative template
+ * to show when the feature is disabled.
+ *
+ * @example
+ * with default template
+ *  ```
+ *  </ng-container *dotShowHideFeature="featureFlag">
+ *       <feature-component></feature-component>
+ *   </ng-container>
+ *  ```
+ *
+ * @example
+ * with default template and alternate template
+ * ```
+ * <ng-template #enabledComponent>
+ *       <feature-component></feature-component>
+ * </ng-template>
+ *
+ * <ng-template #disabledComponent>
+ *       <alternate-component></alternate-component>
+ * </ng-template>
+ *
+ * <ng-container
+ *      *dotShowHideFeature="featureFlag; alternate: disabledComponent"
+ *      [ngTemplateOutlet]="enabledComponent"
+ *  ></ng-container>
+ * ```
+ *
+ * @export
+ * @class DotShowHideFeatureDirective
+ * @implements {OnInit}
+ */
+@Directive({
+    selector: '[dotShowHideFeature]',
+    standalone: true
+})
+export class DotShowHideFeatureDirective implements OnInit {
+    private _featureFlag: string;
+    @Input() set dotShowHideFeature(featureFlag: string) {
+        this._featureFlag = featureFlag;
+    }
+
+    private _alternateTemplateRef: TemplateRef<Component>;
+    @Input() set dotShowHideFeatureAlternate(alternateTemplateRef: TemplateRef<Component>) {
+        this._alternateTemplateRef = alternateTemplateRef;
+    }
+
+    get alternateTemplateRef(): TemplateRef<Component> {
+        return this._alternateTemplateRef;
+    }
+
+    constructor(
+        private templateRef: TemplateRef<Component>,
+        private viewContainer: ViewContainerRef,
+        private dotPropertiesService: DotPropertiesService
+    ) {}
+
+    ngOnInit() {
+        this.dotPropertiesService.getKey(this._featureFlag).subscribe((value) => {
+            const isEnabled = value && value === 'true';
+            this.viewContainer.clear();
+
+            if (isEnabled) {
+                this.viewContainer.createEmbeddedView(this.templateRef);
+            } else if (this.alternateTemplateRef) {
+                this.viewContainer.createEmbeddedView(this.alternateTemplateRef);
+            }
```

Comment:
What happened is `isEnable` is `false` and the dev didn't pass the `this.alternateTemplateRef`?

Reply:
Nothing will be rendered. I've added this conditional so that when we're developing a new component with a `feature flag` we don't have to pass an `Alternate Template` because there isn't going to be one.

Reply:
Should I remove this conditional and make sending an alternate template required?

Reply:
I think we should trow at least a warning in the console.

Reply:
Done!